docs/src/main/asciidoc/mailer-reference.adoc

```diff
@@ -0,0 +1,351 @@
+            new File("quarkus-logo.png"),
+            "image/png", "<my-image@quarkus.io>"));
+}
+----
+
+Note the _content-id_ format and reference.
+By spec, when you create the inline attachment, the content-id must be structured as follows: `<id@domain>`.
+If you don't wrap your content-id between `<>`, it is automatically wrapped for you.
+When you want to reference your attachment, for instance in the `src` attribute, use `cid:id@domain` (without the `<` and `>`).
+
+[[templates]]
+== Message Body Based on Qute Templates
+
+It's possible to inject a mail template, where the message body is created automatically using link:qute[Qute templates].
+
+[source, java]
+----
+@Path("")
+public class MailingResource {
+
+    @CheckedTemplate
+    static class Templates {
+        public static native MailTemplateInstance hello(String name); // <1>
+    }
+
+    @GET
+    @Path("/mail")
+    public Uni<Void> send() {
+        // the template looks like: Hello {name}! // <2>
+        return Templates.hello("John")
+           .to("to@acme.org")  // <3>
+           .subject("Hello from Qute template")
+           .send(); // <4>
+    }
+}
+----
+<1> By convention, the enclosing class name and method names are used to locate the template. In this particular case,
+we will use the `src/main/resources/templates/MailingResource/hello.html` and `src/main/resources/templates/MailingResource/hello.txt` templates to create the message body.
+<2> Set the data used in the template.
+<3> Create a mail template instance and set the recipient.
+<4> `MailTemplate.send()` triggers the rendering and, once finished, sends the e-mail via a `Mailer` instance.
+
+TIP: Injected mail templates are validated during build.
+The build fails if there is no matching template in `src/main/resources/templates`.
+
+You can also do this without type-safe templates:
+
+[source, java]
+----
+@Inject
+@Location("hello")
+MailTemplate hello; // <1>
+
+@GET
+@Path("/mail")
+public Uni<Void> send() {
+    return hello.to("to@acme.org") // <2>
+       .subject("Hello from Qute template")
+       .data("name", "John") // <3>
+       .send() // <4>
+}
+----
+<1> If there is no `@Location` qualifier provided, the field name is used to locate the template.
+Otherwise, search for the template as the specified location. In this particular case, we will use the `src/main/resources/templates/hello.html` and `src/main/resources/templates/hello.txt` templates to create the message body.
+<2> Create a mail template instance and set the recipient.
+<3> Set the data used in the template.
+<4> `MailTemplate.send()` triggers the rendering and, once finished, sends the e-mail via a `Mailer` instance.
+
+
+[[testing]]
+== Testing email sending
+
+Because it is very inconvenient to send emails during development and testing, you can set the `quarkus.mailer.mock` boolean
+configuration to `true` to not actually send emails but print them on stdout and collect them in a `MockMailbox` bean instead.
```

Comment:
```suggestion
configuration to `true` to prevent the actual sending of emails but instead print them on stdout and collect them in a `MockMailbox` bean instead.
```